platform/src/error_code.rs

```diff
@@ -226,3 +230,25 @@ pub enum ErrorCode {
     N01016 =  1016, N01017 =  1017, N01018 =  1018, N01019 =  1019, N01020 =  1020,
     N01021 =  1021, N01022 =  1022, N01023 =  1023,
 }
+
+/// The provided value is not a recognized TRD 104 error code.
+#[derive(PartialEq, Eq)]
+pub struct NotAnErrorCode;
+
+impl fmt::Debug for NotAnErrorCode {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.write_str("not an error code")
+    }
+}
```

Comment:
Why implement `fmt::Debug` manually rather than using `derive(Debug)`?

This should probably print `NotAnErrorCode`, to be consistent with other `Debug` implementations.

Reply:
> Why implement `fmt::Debug` manually rather than using `derive(Debug)`?

Habit in my own code, where I tend to manually `impl Debug`. Switched to a `derive(Debug)`.